console/src/main/java/org/eclipse/rdf4j/console/command/Verify.java

```diff
@@ -118,14 +152,76 @@ public void execute(String... tokens) {
 	}
 
 	/**
-	 * Parse URL or path to local file. Only the second (index 1) token will be parsed, files will be prefixed with
-	 * "file:" scheme
+	 * Validate an RDF data source using a SHACL file or URL, writing the report to a file
 	 * 
-	 * @param tokens array of tokens to be parsed
+	 * @param dataPath  file or URL of the data to be validated
+	 * @param shaclPath file or URL of the SHACL
+	 * @parem reportFile file to write validation report to
+	 */
+	private void shacl(String dataPath, String shaclPath, String reportFile) {
+		ShaclSail sail = new ShaclSail(new MemoryStore());
+		SailRepository repo = new SailRepository(sail);
+		repo.init();
+
+		// load shapes first
+		boolean loaded = false;
+		try {
+			consoleIO.writeln("Loading shapes from " + shaclPath);
+
+			URL shaclURL = new URL(shaclPath);
+			RDFFormat format = Rio.getParserFormatForFileName(shaclPath).orElseThrow(Rio.unsupportedFormat(shaclPath));
+
+			try (SailRepositoryConnection conn = repo.getConnection()) {
```

Comment:
If you call sail.disableValidation() before getting the connection. Then there won't be any validation run on the empty set.

Reply:
Then call sail.enableValidation() after you are done loading the shapes

Reply:
Thanks. Might be interesting to add that the example in the documentation as well http://docs.rdf4j.org/programming/#_validation_with_shacl